### PR TITLE
Tests: use pulsar server in docker

### PR DESCRIPTION
**Pull request by nicoloboschi**
CI was failing with timeouts. At some point tests are stuck in consuming messages that never arrive. 
The culprit might be the JVM not keeping up in running both Pulsar service and the tests. 

In this solution I move out the Pulsar broker to a docker container using testcontainers to isolate resource usage and help debugging

### CODE REVIEW

File: pulsar-jms/pom.xml [hunk 1]
```diff
@@ -88,23 +88,28 @@
...
     </dependency>
     <dependency>
-      <groupId>org.mockito</groupId>
-      <artifactId>mockito-core</artifactId>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>junit-jupiter</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.testcontainers</groupId>
```
Comment (eolivelli) [inline pulsar-jms/pom.xml:111]: is this importing the Pulsar client as a dependency ?

File: pulsar-jms/src/test/java/com/datastax/oss/pulsar/jms/utils/PulsarContainerExtension.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+      network.close();
+    }
+  }
+
+  @Override
+  @SneakyThrows
+  public void beforeAll(ExtensionContext extensionContext) {
+    network = Network.newNetwork();
+    pulsarContainer =
+        new PulsarContainer(DockerImageName.parse("apachepulsar/pulsar:3.0.0"))
```
Comment (eolivelli) [inline pulsar-jms/src/test/java/com/datastax/oss/pulsar/jms/utils/PulsarContainerExtension.java:70]: what about making this a constant ?

**Review by eolivelli — APPROVED** [review]
Overall LGTM

I have left some minor comments

File: pulsar-jms/pom.xml (hunk 1 above)
Comment (nicoloboschi) [reply pulsar-jms/pom.xml:111]: no it's not, it's just the PulsarContainer class that doesn't use any Pulsar lib

File: pulsar-jms/src/test/java/com/datastax/oss/pulsar/jms/VirtualDestinationsConsumerTest.java [hunk 3]
```diff
@@ -465,8 +465,8 @@ public void testPatternConsumerAddingTopicWithServerSideFilters() throws Excepti
             pulsarContainer
                 .getAdmin()
                 .topics()
-                .createSubscription(
-                    topicName, "jms-queue", MessageId.earliest, false, subscriptionProperties);
+                .updateSubscriptionProperties(
```
Comment (eolivelli) [inline pulsar-jms/src/test/java/com/datastax/oss/pulsar/jms/VirtualDestinationsConsumerTest.java:468]: why are we changing this ?

**Review by eolivelli — APPROVED** [review]
LGTM

**Pull request merged**